### ISSUE TITLE
SGD Batch_size = 1 problem

### ISSUE THREAD

**Issue by xiaosean** [jfzhang95/pytorch-deeplab-xception#11]
First, thank you share this cod
e.
I find a problem when I use batch_size = 1

 ```
self.global_avg_pool = nn.Sequential(nn.AdaptiveAvgPool2d((1, 1)),
                                             nn.Conv2d(2048, 256, 1, stride=1, bias=False),
                                             nn.BatchNorm2d(256),
                                             nn.ReLU())
```
Using AdaptiveAvgPool2d will get (1, 2048, 1, 1)
nn.Conv2d => (1, 256, 1, 1)
nn.BatchNorm2d(256) => **problem in here**
Three dimensions be one, BatchNorm2d will raise error

nn.BatchNorm2d(256), #if batch size ==1 should comment out

**Comment by jfzhang95** [comment]
Yes, you can set model.eval() in [here](https://github.com/jfzhang95/pytorch-deeplab-xception/blob/f327b1eb6688ba11faf94b6184dc6264c1f54f8d/networks/deeplab_xception.py#L390) to avoid this error. 
However, in training process, batch_size = 1 will raise this error. Thank you!
